### PR TITLE
Rule S1541: Introduce support for top level statements

### PR DESCRIPTION
**Pull request by dmitry-golovinov-sonarsource**

### CODE REVIEW

File: analyzers/src/SonarAnalyzer.CSharp/Rules/FunctionComplexity.cs [hunk 1]
```diff
@@ -48,6 +49,15 @@ public class FunctionComplexity : ParameterLoadingDiagnosticAnalyzer
 
         protected override void Initialize(ParameterLoadingAnalysisContext context)
         {
+            context.RegisterSyntaxNodeActionInNonGenerated(c =>
+                {
+                    if (c.ContainingSymbol is IMethodSymbol)
```
Comment (dmitry-golovinov-sonarsource) [inline analyzers/src/SonarAnalyzer.CSharp/Rules/FunctionComplexity.cs:54]: Will update this line, when S3776 will be merged

**Review by mary-georgiou-sonarsource — APPROVED** [review]
LGTM

**Review by costin-zaharia-sonarsource — APPROVED** [review]
LGTM! Please don't forget to address [this comment](https://github.com/SonarSource/sonar-dotnet/pull/5099#discussion_r757298291) once https://github.com/SonarSource/sonar-dotnet/pull/5096 is merged.

**Review by costin-zaharia-sonarsource — APPROVED** [review]
LGTM! Please don't forget to address [this comment](https://github.com/SonarSource/sonar-dotnet/pull/5099#discussion_r757298291) once https://github.com/SonarSource/sonar-dotnet/pull/5096 is merged.

**Pull request merged**